pkg/server/apiextensions.go

```diff
@@ -332,7 +321,7 @@ func (c *apiBindingAwareCRDLister) Refresh(crd *apiextensionsv1.CustomResourceDe
 	}
 
 	// Start with a shallow copy
-	refreshed := shallowCopyCRD(updatedCRD)
+	refreshed := shallowCopyCRDAndDeepCopyAnnotations(updatedCRD)
```

Comment:
w00t? why are we modifying a live object? can we ensure no one else is actually using the obj?

Reply:
ic, we acquire a lock